### ISSUE TITLE
2.6 significantly slower than previous versions. 

### ISSUE THREAD

**Issue by yurividal** [rancher/rancher#34641]
**Rancher Server Setup**
Rancher 2.6 Helm, k3s 19

Self Hosted

Hi. I decided to update my rancher instance to the latest 2.6 version.
To my regret, it feels painfully slower than previous 2.5.6

I was using the old UI in 2.5 (cluster manager) , and this update forces me to use the new UI (cluster explorer), which is fine, as long as it's not that slow.

For example: the ingress page takes about 20+ seconds to load. Even when I'm in the same LAN as rancher, managing a cluster that is also on the same lan. Tested Chrome and Firefox.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah, Rancher is pretty much not usable right now for us. We stopped working with Rancher and work only in the Lens Desktop app... :(

**Comment by anupama2501** [comment]
I'm not able to reproduce this issue.

@yurividal would you be able to provide us some more info, we will try to reproduce the same in our setup
the type of clusters  and is it is seen for all or only a few of the clusters
number of ingress or the config used in the ingress
Is it happening only with ingresses or any other services? 
Any network or console logs when the issue happens

**Comment by StrongMonkey** [comment]
Would be great to post pictures of browser dev tool to see which request is taking longer than expected.

**Comment by h0jeZvgoxFepBQ2C** [comment]
It looks like its not the requests itself, but more a javascript problem...  See here, the pod details take ages to load f.e.. 

https://user-images.githubusercontent.com/10075/132401015-687a0a7e-a8ea-4294-b8da-fadb3d77dd53.mp4

**Comment by ineus** [comment]
Chromium is a little bit faster than Firefox, but not too much: like 7 seconds to completely render a deployment's page compared to 10 seconds  in Firefox. So it's not a browser issues but the JS issue indeed

**Comment by jayjensen** [comment]
Same problem here, about 7 to 10 seconds to render pages.
Rancher 2.60 stable on RKE1 (bare metal).
Before the upgrade, everything worked a lot faster (Firefox with same browser version).
(Already rolled back Rancher to 2.5.7, working well again.)

**Comment by synsph** [comment]
Same here ... I user Firefox and have also experienced this same problem, its not useable. 
Shame

**Comment by Jono-SUSE-Rancher** [comment]
@gaktive & @vincent99 - Can you guys take a look at this and let me know if this is something we can look at? I feel like we should have a discussion about this.

cc: @nwmac

**Comment by gaktive** [comment]
@Jono-SUSE-Rancher Vince has already been digging deep into Firefox performance issues for 2.6.x based on feedback elsewhere. 

There are some initial fixes tied to how pages are loaded via proxy here: https://github.com/rancher/dashboard/pull/4153
We also have some improvements in place for look-ups when typing:
https://github.com/rancher/dashboard/issues/4174

This should start to help. We can check with him next week when he's available on more details.

**Comment by pulberg** [comment]
For me on 2.6.0 it's always the websocket connections that get stalled for minutes upon minutes -

![Screen Shot 2021-09-20 at 8 37 03 PM](https://user-images.githubusercontent.com/11069351/134099505-24dab882-afb7-4885-afa6-854c421aa6d6.png)

**Comment by vincent99** [comment]
It's a websocket; it's supposed to stay connected forever.

**Comment by pulberg** [comment]
@vincent99 So it's supposed to stay on "stalled" forever?  I don't see this behavior on my 2.5.x clusters.

**Comment by pulberg** [comment]
Can you tell us what we should be looking at to determine what might be happening?  Also in my case, if I navigate away from the Rancher 2.6 tab for a few min, then go back to the Rancher 2.6 tab, the tab is completely blank, i have to refresh to see any UI elements again, and even when things are back, they are not clickable, everything is frozen.  Then after a few min things are back, but then the cycle starts all over again.

**Comment by vincent99** [comment]
The detail page thing shown in the video above is mostly one small, very inefficient way to figure out if the Conditions tab should be shown and is fixed [here](https://github.com/rancher/dashboard/pull/4242#issuecomment-924524579) (for 2.6.1-rc4+, not in rc3).

@pulberg what you're describing sounds like it's not even running (like an extension that pauses background tabs?) or is so backed up that it can't even render anything (task manager would show 100% or the do you want to kill this tab thing), which I wouldn't expect to just randomly happen after a while of doing nothing (unless maybe there's a huge rate of change events coming in from things happening that it can't keep up with.)  At this point I would say just wait and try 2.6.1 because a lot of the most common-path stuff has been addressed.

**Comment by pulberg** [comment]
@vincent99 Yes, this is what happens, the cpu% goes very high, then i get the "do you want to kill this tab thing", i click "wait", the cycle repeats, pretty much impossible to do any UI things.

![rancher2 6hang](https://user-images.githubusercontent.com/11069351/134546136-626422ab-9b58-460f-8e5c-d0a1f6b9db03.png)

**Comment by gaktive** [comment]
Thanks for all the inputs. As noted earlier, we do have some fixes in place for 2.6.1 but keep this ticket updated with other findings since we may get more in.

**Comment by Neptos** [comment]
I recently upgraded to 2.6.1 from 2.5.something and while the browser tab doesn't become completely unresponsive it still takes ~4seconds to render the list of deployments. All requests are sub 100ms so there is still an issue on the JS side. Performance profiling it claims that the click dom event itself takes almost 1 second to run.

Thanks to @jayjensen for the tip on how to reach the old UI. Saves me from having to downgrade to 2.5 to properly use rancher :+1:

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah, it's a bit better now, but still far away from a "responsive UI".. Hope you can add more performance fixes soon!

**Comment by rajeshneo** [comment]
> Yeah, it's a bit better now, but still far away from a "responsive UI".. Hope you can add more performance fixes soon!

Which one is better? 2.6.2? I have installed 2.6.1 and still a lot of lag.

**Comment by h0jeZvgoxFepBQ2C** [comment]
2.6.2 is much better, but yeah, as @Neptos said, there is still room for improvements imo.

**Comment by gaktive** [comment]
Additional backend improvements are underway here, which are being targeted for 2.6.3: https://github.com/rancher/rancher/issues/25487

**Comment by lenkeybalis** [comment]
Dear all,

Today upgraded my cluster to 1.21.8 version and also upgraded the Rancher version  from 2.6.2 to 2.6.3. For me, the performance of the UI is much better. Also I would like to mention I use Brave as browser.

**Comment by angeloudy** [comment]
I upgraded from 2.5.2 to 2.6.3. Now it's extremely slow!

**Comment by tm-chang** [comment]
We encountered the same issue. We use 2.6.6 now, any progress for this case?

**Comment by KeynesLee** [comment]
We saw the latest version 2.6.7 just released. 
Does it fixed this issue ?  We do not see related info in the release notes.

Is there anyone tried ?

**Comment by gaktive** [comment]
A few improvements have been made for 2.6.7 from the rancher/dashboard side, though they're small "quick wins" that we could get into the release as UI focuses on deeper fixes in upcoming releases.

For those that want, please try that version out. In the UI is now a Diagnostics page to help collect data if you do see continued performance problems.

**Comment by zakariais** [comment]
We are on 2.7.1 version and the UI is extremely slow, we have imported eks cluster, the cluster runs 459 nodes.
Can you suggest some tuning, we can do to improve UI performance?

**Comment by gaktive** [comment]
@zakariais I know you've been chatting with @richard-cox on a few suggestions in Slack but for those reading this ticket, within the UI, under Diagnostics, there are two checkboxes to try as a workaround: _Incremental Loading_ and _Manual Refresh_.

**Comment by konvergence** [comment]
We have the same issue (see https://forums.rancher.com/t/rancher-started-work-extremely-slow/38790/2)
Specially on pods view.

Even if we have selected namespaces, rancher UI load all pods infos on browser side and filter display.
It result a  high memory/cpu usage on the browser, that stuck the browser (see https://github.com/rancher/rancher/issues/34641#issuecomment-925967124)

Why rancher UI don't really load pods infos only on selected namespaces ?

**Comment by richard-cox** [comment]
There's been a large number of performance improvements since this issue was created. 

We would advise everyone to update to latest Rancher, in particular the soon to be released 2.7 Q2 version.

A lot of the performance improvements UI side can be enabled via the burger menu --> `Global Settings` --> `Performances` settings. In the upcoming 2.7 Q2 release there are two in particular we'd ask users to try. 

1) Websocket Web Worker
Updates to resources pushed to the UI come WebSockets and are handled in the UI thread. Enable this option to handle cluster WebSocket updates in a Web Worker in a separate thread. This should help the responsiveness of the UI in systems where resources change often.

2) Require Namespace / Project Filtering
Require the user to select namespaces and/or projects when viewing cluster resources. This restricts the number of resources fetched when viewing lists and should help the responsiveness of the UI in systems with a lot of resources.

We'd love to receive feedback on any of these settings, whether they assist or don't. That feedback will help us direct the next iteration of performance improvements.

If these improvements don't help, we'd love to know more detail. If there is slowness where in the dashboard is worst affected? If the issue is when viewing a downtream cluster what kind of resource counts are there? Are there any failed network requested, errors in the console or message churn in the web sockets (all available via the browsers Dev Tools interface)?
